File: Iniciante/1041 - Coordenadas de um Ponto/TypeScript/1041.ts
import * as fs from "fs";

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const lines: string[] = input.trim().split(/\s+/);
  const [x, y] = lines.map(Number);
  const location: string = determineLocation(x, y);
  console.log(location);
}

function determineLocation(x: number, y: number): string {
  if (x === 0 && y === 0) {
    return "Origem";
  } else if (x === 0) {
    return "Eixo Y";
  } else if (y === 0) {
    return "Eixo X";
  } else if (x > 0 && y > 0) {
    return "Q1";
  } else if (x < 0 && y > 0) {
    return "Q2";
  } else if (x < 0 && y < 0) {
    return "Q3";
  } else {
    return "Q4";
  }
}

main();

File: Iniciante/1041 - Coordenadas de um Ponto/Python/1041.py
def main():
    x, y = map(float, input().split())
    quadrant_status = check_quandrant_status(x, y)
    print(quadrant_status)


def check_quandrant_status(x, y):
    if x == 0 and y == 0:
        return "Origem"
    elif x == 0:
        return "Eixo Y"
    elif y == 0:
        return "Eixo X"
    elif x > 0 and y > 0:
        return "Q1"
    elif x < 0 and y > 0:
        return "Q2"
    elif x < 0 and y < 0:
        return "Q3"
    else:
        return "Q4"


if __name__ == "__main__":
    main()
